### PR TITLE
Omit WooCommerce plugin from the list of Woo extensions in Helper

### PR DESCRIPTION
**Pull request by andfinally**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

p1696252501678409-slack-C03Q87XT1QF

* Woo Express relied on the `woocommerce_show_addons_page` to hide the current addons page. The filter was removed in an earlier PR. This PR restores it: if the filter returns false, we won't add the addons page.
* The Woo release team uses the product 18734002369816 to distribute the WooCommerce weekly release to Atomic sites. (See pcShBQ-14K-p2). To do this, the plugin needs a `Woo: ` header in the bootstrap file. This means that Helper [identifies it as a Woo extension](https://github.com/woocommerce/woocommerce/blob/de1fe1bf96e10a8e6581d36851574a9844d62efb/plugins/woocommerce/includes/admin/helper/class-wc-helper.php#L1170). It's then listed as a plugin which doesn't have a corresponding WooCommerce.com subscription. To hide it in the Helper list, I've added a condition which omits the plugin if (1) it has a `Woo: ` header (2) its name is `WooCommerce`.
* During the transition to a React-based marketplace, we're temporarily [hiding](https://github.com/woocommerce/woocommerce/pull/39751) an extra submenu item using JS. This PR uses a better method of hiding the item,  borrowed from [Jetpack](https://github.com/Automattic/jetpack/blob/cf34b99ab8418ad936dedd08bd73f790b4f9180d/projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php#L320).

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Reproducing the problem

1. Instead of trying this in your local WooCommerce `wp-env` environment, create a JN or other test WordPress site and install a WooCommerce weekly release like this one: 
[woocommerce-8.2.0.2.zip](https://github.com/woocommerce/woocommerce/files/12784123/woocommerce-8.2.0.2.zip) (You can also download this from https://woocommerce.com/my-account/downloads/: it's listed as `WooCommerce (weekly release)`.)
2. Go to `wp-admin/admin.php?page=wc-addons&path&section=helper` and connect your test site to your WooCommerce.com account, by clicking on the "Connect" button in the screen shown below. When connected, the page will refresh to show a list of your Woo extensions. Notice that WooCommerce is listed as a "Woo extension" without a subscription, like in the other screenshot below.
3. Deactivate and delete that version of the WooCommerce plugin.

#### Testing the fix for the WooCommerce plugin

5. Check out this branch.
6. When run locally, the zip build doesn't have enough RAM to generate the translation files. Comment out these two lines in `plugins/woocommerce/bin/build-zip.sh` to skip that step:

```diff
Index: plugins/woocommerce/bin/build-zip.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/plugins/woocommerce/bin/build-zip.sh b/plugins/woocommerce/bin/build-zip.sh
--- a/plugins/woocommerce/bin/build-zip.sh	(revision ceafa7da30b0211621222792bfe679b21e8a8c3c)
+++ b/plugins/woocommerce/bin/build-zip.sh	(date 1696261507943)
@@ -18,8 +18,8 @@
 pnpm -w run build --filter=woocommerce || exit "$?"
 echo "Cleaning up PHP dependencies..."
 composer install --no-dev || exit "$?"
-echo "Run makepot..."
-pnpm -r --filter=woocommerce run makepot || exit "$?"
+#echo "Run makepot..."
+#pnpm -r --filter=woocommerce run makepot || exit "$?"
 echo "Syncing files..."
 rsync -rc --exclude-from="$PROJECT_PATH/.distignore" "$PROJECT_PATH/" "$DEST_PATH/" --delete --delete-excluded
```

7. Now edit `plugins/woocommerce/woocommerce.php` and add the `Woo` header `* Woo: 18734002369816:624a1b9ba2fe66bb06d84bcdd401c6a6` to the plugin header. The header should look like this. (Don't worry about the version number, it's isn't important here.)

```php
/**
 * Plugin Name: WooCommerce
 * Plugin URI: https://woocommerce.com/
 * Description: An eCommerce toolkit that helps you sell anything. Beautifully.
 * Version: 8.2.0.2
 * Author: Automattic
 * Author URI: https://woocommerce.com
 * Text Domain: woocommerce
 * Domain Path: /i18n/languages/
 * Requires at least: 6.2
 * Requires PHP: 7.4
 *
 * Woo: 18734002369816:624a1b9ba2fe66bb06d84bcdd401c6a6
 *
 * @package WooCommerce
 */
```

8. Build the zip file so you can install this branch: `cd plugins/woocommerce && ./bin/build-zip.sh`.
9. Upload the zip file from `plugins/woocommerce` to your test site and enable the plugin.
10. Go back to the helper page. Notice that WooCommerce is no longer listed.

#### Testing the hidden submenu item

Inspect the HTML after the Extensions submenu item in the WooCommerce menu. There should be a hidden item with the `hide-if-js` class.

image

If you don't see this item, with the new marketplace UI, enable the `Marketplace` feature in `WooCommerce > Settings > Advanced > Features`.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Adds condition to ensure WooCommerce is not listed as a Woo extension in the Helper list. Restores the `woocommerce_show_addons_page` filter as a means of controlling whether the addons page is added as a WooCommerce submenu item. Hides a temporary extra addons submenu item using a better method borrowed from Jetpack.

#### Comment

</details>

#### Screenshots

image

### CODE REVIEW

**Review by kdevnel — APPROVED** [review]
Thanks for this!

File: plugins/woocommerce/includes/admin/class-wc-admin-menus.php [hunk 1]
```diff
@@ -40,13 +47,16 @@ public function __construct() {
...
 		add_action( 'admin_menu', array( $this, 'status_menu' ), 60 );
 
-		if ( FeaturesUtil::feature_is_enabled( 'marketplace' ) ) {
-			$container = wc_get_container();
-			$container->get( Marketplace::class );
+		// Woo Express uses this filter!
+		if ( apply_filters( 'woocommerce_show_addons_page', true ) ) {
+			if ( FeaturesUtil::feature_is_enabled( 'marketplace' ) ) {
+				$container = wc_get_container();
+				$container->get( Marketplace::class );
```
Comment (xristos3490) [inline plugins/woocommerce/includes/admin/class-wc-admin-menus.php:61]: Nit: Is this needed?

**Review by xristos3490 — APPROVED** [review]
Looks good! Re-adding the `woocommerce_show_addons_page` hook can assist WX and other third-party developers who were previously using this filter!

Added a minor comment! Pre-approving!

File: plugins/woocommerce/includes/admin/class-wc-admin-menus.php (hunk 1 above)
Comment (andfinally) [reply plugins/woocommerce/includes/admin/class-wc-admin-menus.php:61]: According to https://github.com/woocommerce/woocommerce/blob/32e01a21f2b7c5a0b9cfa87b9d75488594dd8b91/plugins/woocommerce/includes/README.md#interacting-with-the-src-folder, it's the new way of getting an instance of a class from the `src` folder.

**Pull request merged**